troubleshooting (#3250)
hashicorp/web-unified-docs: content/terraform-docs-common/docs/cloud-docs/scim/troubleshooting.mdx

Reviewer: This one is a TFE only event.

Reviewer: This is also only available in TFE

---
page_title: Troubleshoot SCIM provisioning
description: >-
  Diagnose and resolve common issues with SCIM provisioning, including authentication failures, validation errors, team linking
  conflicts, and rate limits.
# START AUTO GENERATED METADATA, DO NOT EDIT
created_at: 2026-08-05T15:05:42-07:00
last_modified: 2026-08-28T22:24:38.000Z
# END AUTO GENERATED METADATA
---

# Troubleshoot SCIM provisioning

This topic describes troubleshooting steps for diagnosing and resolving issues with SCIM provisioning in <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name -->.

> Use this page for <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise-side<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform-side<!-- END: TFC:only name:product-name --> behavior, SCIM responses,<!-- BEGIN: TFEnterprise:only name:page-scope --> team linking<!-- END: TFEnterprise:only name:page-scope -->, and operational follow-up. 

Reviewer: This is rendering with a double comma on the preview.

If a failure occurs before <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name --> receives the request, check your identity provider provisioning logs first.

For provider-specific setup, assignment, scope, and attribute-mapping checks, use the [configure guides](/terraform/cloud-docs/scim/configure) after these Terraform Enterprise-side checks.

> ## Escalation path

Reviewer: We may want to have this entire escalation path section under TFE only. My thinking is that checking the IdP is already covered in the section above, support bundles are TFE only, and I don't know if we typically use audit events for debugging purposes.

To resume SCIM synchronization, open **Manage** and click **Resume SCIM provisioning**.

Resuming site-level SCIM does not replay or backfill changes that your identity provider did not send while SCIM was paused. Terraform Enterprise resumes processing only new SCIM requests after you resume SCIM provisioning.
<!-- END: TFEnterprise:only name:pause-site -->

<!-- BEGIN: TFC:only name:pause-site -->

> ### Disable SCIM for the organization

Reviewer: ```suggestion
```

I thought removing this header would make it more clear that disabling=pausing

Disabling SCIM at the organization level stops all SCIM provisioning requests for your HCP Terraform organization.

HCP Terraform organization owner steps:

1. Navigate to **Organization settings** > **SCIM provisioning**.
1. Click **Manage**.
1. Click **Disable SCIM**.
1. Confirm the action.

For more information about disabling SCIM, refer to [Manage SCIM provisioning](/terraform/cloud-docs/scim/manage#disable-scim-provisioning).

While SCIM is disabled:

- Changes to users and groups in your IdP are not synchronized to HCP Terraform.
- Existing provisioned users and teams do not change.
- Users can continue to log in using SSO, including SCIM-managed users.

To re-enable SCIM provisioning, navigate to **Organization settings** > **SCIM provisioning** and click **Enable SCIM**.

> Re-enabling SCIM does not automatically replay changes that your identity provider made while SCIM was disabled. Restart provisioning in Entra ID, or remove and re-add group assignments in Okta, to synchronize any missed changes.

Reviewer: ```suggestion
Re-enabling SCIM does not automatically replay changes that your identity provider made while SCIM was disabled. Restart provisioning in Entra ID, or remove and re-add assignments in Okta, to synchronize any missed changes.
```

Reviewer: It may also help to clarify that when SCIM is disabled, HCPT will reject any requests made by the IdP with an error. The IdP will then report errors with provisioning, which is to be expected.

- Direct membership changes in Terraform Enterprise remain blocked while the link exists. To manage membership manually, unlink the team instead. Refer to [Link SCIM groups to teams](/terraform/enterprise/scim/manage/team-mapping).

When you resume synchronization, Terraform Enterprise immediately reconciles the team's membership with the current SCIM group state.
<!-- END: TFEnterprise:only name:pause-team -->

## Check HTTP status codes

You can troubleshoot issues by cross-referencing error code information with your SCIM implementation. For status codes, validation rules, and request details for each endpoint family, refer to the corresponding API reference:

<!-- BEGIN: TFC:only name:endpoint-api-refs -->

> - [Public SCIM API](/terraform/cloud-docs/api-docs/scim)

Reviewer: ```suggestion
- [SCIM API](/terraform/cloud-docs/api-docs/scim)
```

The term "public" makes more sense in the context of TFE.

1. Check the `last-used-at` timestamp to confirm the IdP is successfully authenticating.
1. If necessary, create a new token. SCIM token values are displayed only once when created. If you lose the token value, create a new token and rotate the IdP to use it.
<!-- END: TFEnterprise:only name:401-admin-steps -->

<!-- BEGIN: TFC:only name:401-admin-steps -->
1. Navigate to **Organization settings** > **SCIM provisioning** > **Tokens**.
1. Verify that an active token exists and has not expired.
1. If necessary, generate a new token. Set an expiration date between 30 days and 1 year. SCIM token values are displayed only once when generated. If you lose the token value, generate a new token and update the IdP to use it.
<!-- END: TFC:only name:401-admin-steps -->

IdP administrator:

After the <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise site admin<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform organization owner<!-- END: TFC:only name:product-name --> provides a new token, update the IdP configuration to use the new token and rerun the provisioning test.

### `403 Forbidden` error

> HTTP `403` errors returned by the public SCIM provisioning endpoints, `/scim/v2/Users` or `/scim/v2/Groups`, usually mean <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name --> is not currently accepting SCIM provisioning requests.

Reviewer: ```suggestion
HTTP `403` errors returned by the SCIM provisioning endpoints, `/scim/v2/Users` or `/scim/v2/Groups`, usually mean <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name --> is not currently accepting SCIM provisioning requests.
```

- The IdP is not sending a `userName` that already exists for another SCIM identity. That condition returns `409 Conflict`.
- The IdP request includes at least one email entry. A primary email is preferred, but Terraform Enterprise can also use the first email entry if none is marked primary. Missing email data returns `400 Bad Request`.
- If Terraform Enterprise already has a manually created user with the same email address, verify that the IdP is sending the same email address for that user so linking can occur.
<!-- END: TFEnterprise:only name:users-syncing-conditions -->

<!-- BEGIN: TFC:only name:users-syncing-conditions -->
Verify that the user is assigned or otherwise in scope for provisioning in your identity provider.
<!-- END: TFC:only name:users-syncing-conditions -->
<!-- BEGIN: TFEnterprise:only name:duplicate-username -->
If the issue is a duplicate `userName`:

1. The Terraform Enterprise site admin identifies the existing SCIM-managed user that already owns that `userName`.
1. The IdP administrator renames the user in the IdP or corrects the conflicting SCIM identity before retrying.
1. Retry provisioning.
<!-- END: TFEnterprise:only name:duplicate-username -->

> 

Reviewer: I think we could have a separate header for the duplicate user issue to separate it from the user not syncing scenario.

<!-- BEGIN: TFEnterprise:only name:deactivation-behavior -->
Terraform Enterprise differentiates between SCIM deactivation and deletion:

- When your IdP sends the `active=false` attribute, Terraform Enterprise suspends the user.
- When your IdP makes a `DELETE` request to the `/scim/v2/Users/:id` endpoint, Terraform Enterprise removes the SCIM identity and suspends the user.
- If the affected user is a SCIM-provisioned site administrator, Terraform Enterprise revokes site admin access.

If the user remains active in Terraform Enterprise, confirm that your IdP is sending either `active=false` or a SCIM delete request, not only removing the user from a group.
<!-- END: TFEnterprise:only name:deactivation-behavior -->

<!-- BEGIN: TFC:only name:deactivation-behavior -->
HCP Terraform differentiates between SCIM deactivation and deletion. The exact behavior depends on your IdP:

- When your IdP deactivates a user, HCP Terraform sets the user status to inactive and removes their ability to log in. Okta typically sends an inactive update immediately. Entra ID initially sets the user to inactive, then may send a delete request after 30 days.

> - When your IdP sends a `DELETE` request, HCP Terraform removes the user from the organization and deletes the account if it is an organization-owned user.

Reviewer: ```suggestion
- When your IdP sends a `DELETE` request, HCP Terraform removes the user from the organization if it is a user-owned account and deletes the user if it is an organization-owned account.
```
I switched around user and account mostly to avoid the term "user-owned user".

Check for the following conditions:

- The group is configured for provisioning in your identity provider and is in scope for synchronization to <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name -->.
- Group membership updates reference <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name --> SCIM user IDs for users that were already provisioned through `/scim/v2/Users`.
- The IdP is not trying to create or rename a group to a `displayName` that already exists. That condition returns `409 Conflict`.

If the issue is a duplicate group name, the IdP administrator renames or removes the conflicting SCIM group in the identity provider<!-- BEGIN: TFEnterprise:only name:duplicate-group -->

>  before retrying.<!-- END: TFEnterprise:only name:duplicate-group --><!-- BEGIN: TFC:only name:duplicate-group -->, then re-triggers synchronization.

Reviewer: ```suggestion
If the issue is a duplicate group name, rename or remove the conflicting SCIM group in the identity provider<!-- BEGIN: TFEnterprise:only name:duplicate-group -->
 before retrying.<!-- END: TFEnterprise:only name:duplicate-group --><!-- BEGIN: TFC:only name:duplicate-group -->, then re-trigger synchronization.
```

> HCP Terraform team names have a maximum length of 90 characters. If an IdP group name exceeds 90 characters, SCIM cannot create the corresponding team. Rename the IdP group to 90 characters or fewer before retrying.

Reviewer: ```suggestion
HCP Terraform team names have a maximum length of 90 characters. If an IdP group name exceeds 90 characters, HCP Terraform cannot create the corresponding team. Rename the IdP group to 90 characters or fewer before retrying.
```

Admin SCIM `429` responses use the standard admin API error format.

For exact rate-limit values and response details, refer to the [Public SCIM API](/terraform/enterprise/api-docs/scim), the [SCIM Settings API](/terraform/enterprise/api-docs/admin/scim-settings), and the [Team SCIM Group Mapping API](/terraform/enterprise/api-docs/admin/team-scim-group-mapping).
<!-- END: TFEnterprise:only name:429-rate-limits -->

<!-- BEGIN: TFC:only name:429-rate-limits -->
TODO - verify scim api docs

> For public SCIM `/scim/v2/*`, use the `Retry-After` header to determine how long to wait before retrying. Refer to the [public SCIM API](/terraform/cloud-docs/api-docs/scim) for for exact rate-limit values and response details.

Reviewer: ```suggestion
For SCIM `/scim/v2/*`, use the `Retry-After` header to determine how long to wait before retrying. Refer to the [SCIM API](/terraform/cloud-docs/api-docs/scim) for for exact rate-limit values and response details.
```

<!-- END: TFEnterprise:only name:team-sync-fields -->

<!-- BEGIN: TFC:only name:team-sync-statuses -->
The following table describes the team statuses that indicate the SCIM provisioning state:

| Status | Description |
|--------|-------------|
| `Synced` | SCIM is enabled, and the team is in sync with your IdP. |
| Not applicable | SCIM has not been enabled, or this team was manually created and is not managed by SCIM. |
<!-- END: TFC:only name:team-sync-statuses -->

<!-- BEGIN: TFEnterprise:only name:site-admin-group -->
## Site admin group issues

If site admin access is not changing as expected, verify that the correct site admin group is configured in **Site Admin** > **SCIM** and that the relevant group membership update or SCIM settings change completed successfully.

For the site admin group behavior, including grant, revoke, remap, and clear semantics, refer to [Manage SCIM provisioning](/terraform/enterprise/scim/manage#site-admin-group-behavior).

Deleting or disabling SCIM does not automatically revoke site admin access that was already granted.
<!-- END: TFEnterprise:only name:site-admin-group -->

> ## Audit logs

Reviewer: These are "Audit events" instead of logs In HCPT since we don't yet have audit log streaming.

> For compliance and troubleshooting purposes, <!-- BEGIN: TFEnterprise:only name:product-name -->Terraform Enterprise<!-- END: TFEnterprise:only name:product-name --><!-- BEGIN: TFC:only name:product-name -->HCP Terraform<!-- END: TFC:only name:product-name --> logs SCIM operations. SCIM-related audit log entries that include:

Reviewer: Instead of audit logs here for HCPT, this should be audit events.

- Verify that IdP changes are being applied correctly.
- Debug issues by reviewing the sequence of operations.

<!-- BEGIN: TFEnterprise:only name:audit-log-link -->
To inspect audit logs and service logs, refer to [Logs](/terraform/cloud-docs/deploy/manage/monitor/logs).
<!-- END: TFEnterprise:only name:audit-log-link -->

<!-- BEGIN: TFC:only name:audit-log-link -->
To inspect audit logs, refer to [Audit logging](/terraform/cloud-docs/api-docs/audit-trails).
<!-- END: TFC:only name:audit-log-link -->

> ## Disable SCIM

Reviewer: In HCPT, this should be Delete SCIM configuration. There's a difference in terms between HCPT and TFE because we chose to match the the terms used for HCPT SSO configurations, so:

Temporarily stopping SCIM syncing:
- TFE: Pausing
- HCPT: Disabling

Permanently stopping SCIM until it is reconfigured completely:
 - TFE: Disabling
 - HCPT: Deleting the SCIM configuration

<Warning>

> Disabling SCIM is destructive. <!-- BEGIN: TFEnterprise:only name:disable-warning -->If you enable SCIM again later, you must recreate SCIM tokens, reprovision groups, and recreate team mappings. Use it only as a last resort after capturing exact timestamps, relevant IdP log entries, Terraform Enterprise audit or service logs, and a support bundle for further investigation.<!-- END: TFEnterprise:only name:disable-warning --><!-- BEGIN: TFC:only name:disable-warning -->If you re-enable SCIM, you must restart provisioning in your IdP to synchronize any changes made while SCIM was disabled. Use it only as a last resort after capturing exact timestamps, relevant IdP log entries, and HCP Terraform audit logs for further investigation.<!-- END: TFC:only name:disable-warning -->

Reviewer: "recreate team mappings" is a TFE only step.

We should also mention that users must be resynced as well since this is only referring to groups.
